File: ska_frontend/src/modules/global_state/env_config.rs
#[derive(Clone, PartialEq, PartialOrd, Eq, Ord, Debug, Hash)]
pub struct EnvConfig {
    pub env_profile: String,
    pub base_href: String,
    pub frontend_url: String,
    pub backend_url: String,
    pub auth_issuer_url: String,
    pub auth_client_id: String,
    pub auth_client_secret: String,
}

impl EnvConfig {
    pub fn from_env() -> EnvConfig {
        let env_profile = String::from(env!("THAPO_SKA_ENV_PROFILE"));
        let base_href = String::from(env!("THAPO_SKA_BASE_HREF"));
        let frontend_url = String::from(env!("THAPO_SKA_FRONTEND_URL"));
        let backend_url = String::from(env!("THAPO_SKA_BACKEND_URL"));
        let auth_issuer_url = String::from(env!("THAPO_SKA_AUTH_ISSUER_URL"));
        let auth_client_id = String::from(env!("THAPO_SKA_AUTH_CLIENT_ID"));
        let auth_client_secret = String::from(env!("THAPO_SKA_AUTH_CLIENT_SECRET"));
        EnvConfig {
            env_profile,
            base_href,
            frontend_url,
            backend_url,
            auth_issuer_url,
            auth_client_id,
            auth_client_secret,
        }
    }
}

File: ska_frontend2/src/domain/global_state/env_config.ts
export class EnvConfig {
    readonly envProfile: string;
    readonly baseUrl: string;
    readonly frontendUrl: string;
    readonly backendUrl: string;
    readonly authIssuerUrl: string;
    readonly authClientId: string;

    constructor(obj: {
        envProfile: string;
        baseUrl: string;
        frontendUrl: string;
        backendUrl: string;
        authIssuerUrl: string;
        authClientId: string;
    }) {
        this.envProfile = obj.envProfile;
        this.baseUrl = obj.baseUrl;
        this.frontendUrl = obj.frontendUrl;
        this.backendUrl = obj.backendUrl;
        this.authIssuerUrl = obj.authIssuerUrl;
        this.authClientId = obj.authClientId;
    }

    static fromEnv(importMetaEnv: ImportMetaEnv): EnvConfig {
        const envProfile: string = importMetaEnv.VITE_THAPO_SKA_ENV_PROFILE;
        const baseUrl: string = importMetaEnv.BASE_URL;
        const frontendUrl: string = importMetaEnv.VITE_THAPO_SKA_FRONTEND_URL;
        const backendUrl: string = importMetaEnv.VITE_THAPO_SKA_BACKEND_URL;
        const authIssuerUrl: string = importMetaEnv.VITE_THAPO_SKA_AUTH_ISSUER_URL;
        const authClientId: string = importMetaEnv.VITE_THAPO_SKA_AUTH_CLIENT_ID;
        return new EnvConfig({
            envProfile,
            baseUrl,
            frontendUrl,
            backendUrl,
            authIssuerUrl,
            authClientId
        });

    }
}
